lib/jekyll/filters/url_filters.rb

```diff
@@ -23,14 +22,23 @@ def absolute_url(input)
       # Returns a URL relative to the domain root as a String.
       def relative_url(input)
         return if input.nil?
-        site = @context.registers[:site]
-        return ensure_leading_slash(input.to_s) if site.config["baseurl"].nil?
+        return ensure_leading_slash(input.to_s) if sanitized_baseurl.nil?
         Addressable::URI.parse(
-          ensure_leading_slash(site.config["baseurl"]) + ensure_leading_slash(input.to_s)
+          ensure_leading_slash(sanitized_baseurl) + ensure_leading_slash(input.to_s)
         ).normalize.to_s
       end
 
       private
+
+      def site
+        @context.registers[:site]
+      end
+
+      def sanitized_baseurl
+        baseurl = site.config["baseurl"]
+        return baseurl unless baseurl.nil? || (baseurl == "/")
```

Comment:
> What happens if baseurl is set to /foo/bar/ or even something really silly like "///////" or /foo/bar//" etc?

Pray, Mr. Babbage, if you put into the machine wrong figures, will the right answers come out?